creator-node/src/fileManager.js

```diff
@@ -83,143 +83,151 @@ async function saveFileToIPFSFromFS (req, srcPath) {
+        req.logger.debug(`Attempting to fetch multihash ${multihash} by racing replica set endpoints`)
+
+        // Note - Requests are intentionally not parallel to minimize additional load on gateways
+        for (let index = 0; index < gatewayUrlsMapped.length; index++) {
+          const url = gatewayUrlsMapped[index]
+          try {
+            const resp = await axios({
+              method: 'get',
+              url,
+              responseType: 'stream',
+              timeout: 20000 /* 20 sec - higher timeout to allow enough time to fetch copy320 */
+            })
+            if (resp.data) {
+              response = resp
+              break
+            }
+          } catch (e) {
+            req.logger.error(`Error fetching file from other cnode ${url} ${e.message}`)
+            continue
           }
-        } catch (e) {
-          req.logger.error(`Error fetching file from other cnode ${url} ${e.message}`)
-          continue
         }
-      }
 
-      if (!response || !response.data) {
-        throw new Error(`Couldn't find files on other creator nodes`)
-      }
+        if (!response || !response.data) {
+          throw new Error(`Couldn't find files on other creator nodes`)
+        }
 
-      // Write file to disk
-      await Utils.writeStreamToFileSystem(response.data, expectedStoragePath)
-      fileFound = true
+        // Write file to disk
+        await Utils.writeStreamToFileSystem(response.data, expectedStoragePath)
+        fileFound = true
 
-      req.logger.info(`saveFileForMultihash - wrote file to ${expectedStoragePath}`)
-    } catch (e) {
-      throw new Error(`Failed to retrieve file for multihash ${multihash} from other creator node gateways: ${e.message}`)
+        req.logger.info(`wrote file to ${expectedStoragePath}`)
+      } catch (e) {
+        throw new Error(`Failed to retrieve file for multihash ${multihash} from other creator node gateways: ${e.message}`)
+      }
     }
-  }
 
-  // file was not found on ipfs or any gateway
-  if (!fileFound) {
-    throw new Error(`Failed to retrieve file for multihash ${multihash} after trying ipfs & other creator node gateways`)
-  }
+    // file was not found on ipfs or any gateway
+    if (!fileFound) {
+      throw new Error(`Failed to retrieve file for multihash ${multihash} after trying ipfs & other creator node gateways`)
+    }
 
-  // for verification purposes - don't delete. verifies that the contents of the file match the file's cid
-  const ipfs = req.app.get('ipfsLatestAPI')
-  const content = fs.createReadStream(expectedStoragePath)
-  for await (const result of ipfs.add(content, { onlyHash: true, timeout: 2000 })) {
-    if (multihash !== result.cid.toString()) {
-      throw new Error(`File contents don't match IPFS hash multihash: ${multihash} result: ${result.cid.toString()}`)
+    // for verification purposes - don't delete. verifies that the contents of the file match the file's cid
```

Comment:
the other change here is wrapping this code in a try-catch

Reply:
not clear why this sub flow here where we do ipfs add needs its own dedicated try catch? bc `File contents don't match IPFS hash multihash: ${multihash} result: ${result.cid.toString()}` will still be surfaced in the outer catch

Reply:
resolved afk